core/src/main/java/org/jdbi/v3/core/result/LinkedHashMapRowReducer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.result;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.stream.Stream;
+
+/**
+ * A row reducer that uses {@link LinkedHashMap} (which preserves insertion order) as a
+ * result container, and returns {@code map.values().stream()} as the final result.
+ *
+ * <p>Implementors need only implement the {@link #accumulate(Object, RowView)} method.
+ *
+ * @param <K> the map key type--often the primary key type of {@code <V>}.
+ * @param <V> the map value type, and the result element type--often the "master" object in a
+ *     master/detail relation.
+ */
+public abstract class LinkedHashMapRowReducer<K, V> implements RowReducer<Map<K, V>, V> {
```

Comment:
could this be an `@FunctionalInterface` with default methods provided for `container` and `stream`?  is that nicer?

Reply:
My instinct is that it wouldn't buy us anything, but I'll try it out and check.

The issue is that `ResultBearing.reduceRows()` accepts `RowReducer<C, R>`, not this subtype. So the compiler probably wouldn't know what to do with that lambda without casting to `LinkedHashMapRowReducer`.

Alternatively, we could add a `reduceRows(BiConsumer<Map<K, V>, RowView)` overload that used that reducer under the hood.

Reply:
Probably wouldn't hurt anything though.

Reply:
Usage after making it a functional interface, and adding another convenience method in `ResultBearing`:

```java
List<SomethingWithLocations> result = dbRule.getSharedHandle()
    .createQuery("SELECT something.id, name, location FROM something NATURAL JOIN something_location")
    .<Integer, SomethingWithLocations>reduceRows((map, rv) ->
        map.computeIfAbsent(rv.getColumn("id", Integer.class),
                            id -> new SomethingWithLocations(rv.getRow(Something.class)))
           .locations
           .add(rv.getColumn("location", String.class)))
    .collect(toList());
```